pkg/endpoint/bpf.go

```diff
@@ -48,11 +48,12 @@ import (
 )
 
 const (
-	// ExecTimeout is the execution timeout to use in join_ep.sh executions
-	ExecTimeout = 300 * time.Second
-
-	// EndpointGenerationTimeout specifies timeout for proxy completion context
-	EndpointGenerationTimeout = 55 * time.Second
+	// EndpointGenerationTimeout specifies timeout for the whole
+	// endpoint generation process, including the time waiting for:
+	// - locks
+	// - proxy ACKs/NACKs
+	// - BPF recompilation
+	EndpointGenerationTimeout = 300 * time.Second
```

Comment:
Won't this now cause API clients attempting to apply changes to time out now when generation takes too long? This is why we had 55s before, to keep under the common client timeout of 60s.

Reply:
I'll change this to 55 to retain the earlier overall timeout here. I was not aware of the 60s API timeout, and saw that we had a 300s timeout defined for endpoint generation which was never honored due to the 55 sec timeout was already in effect.

Reply:
Sorry the sloppy language, 300s was the `ExecTimeout`